util/src_util/pt_gcry_base.c

```diff
@@ -0,0 +1,206 @@
+/* Copyright (C) 2009 - 2022 National Aeronautics and Space Administration.
+   All Foreign Rights are Reserved to the U.S. Government.
+
+   This software is provided "as is" without any warranty of any kind, either expressed, implied, or statutory,
+   including, but not limited to, any warranty that the software will conform to specifications, any implied warranties
+   of merchantability, fitness for a particular purpose, and freedom from infringement, and any warranty that the
+   documentation will conform to the program, or any warranty that the software will be error free.
+
+   In no event shall NASA be liable for any damages, including, but not limited to direct, indirect, special or
+   consequential damages, arising out of, resulting from, or in any way connected with the software or its
+   documentation, whether or not based upon warranty, contract, tort or otherwise, and whether or not loss was sustained
+   from, or arose out of the results of, or use of, the software, documentation or services provided hereunder.
+
+   ITC Team
+   NASA IV&V
+   jstar-development-team@mail.nasa.gov
+*/
+
+/**
+ *  Unit Tests that make use of TC_ApplySecurity/TC_ProcessSecurity function on the data with KMC Crypto Service/MariaDB Functionality Enabled.
+ *  BE SURE TO HAVE APPROPRIATE SA's READY FOR EACH SET OF TESTS
+ **/
+
+#include "utest.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <gcrypt.h>
+
+#include <time.h>
+#include <unistd.h>
+
+#include "crypto.h"
+#include "crypto_error.h"
+
+#include "shared_util.h"
+
+void random_big_buffer(char* buffer, int32_t buffer_len)
```

Comment:
Should `random_big_buffer` be moved to another file for reuse across the entire project?

Reply:
That's another one, that I was just creating for the purposes of these tests, but if so desired, we can indeed move it to a shared util.